store/src/java/com/zimbra/cs/index/ZimbraHit.java

```diff
@@ -379,7 +380,21 @@ else if (result < 0)
      */
     public static int getReadStatus(ZimbraHit zh) throws ServiceException {
         if (zh instanceof ProxiedHit) {
-            return ((ProxiedHit) zh).getElement().getAttributeInt(MailConstants.A_UNREAD);
+            try {
+                return ((ProxiedHit) zh).getElement().getAttributeInt(MailConstants.A_UNREAD);
+            } catch (ServiceException e) {
+                // This is  message hit
+                Element msgHit = null;
+                try {
```

Comment:
can we move this try block itself in its parent try block at line 383 instead of handling this in catch block

Reply:
no because for some messages you do get u="0" only when we do not get u=0 we fallback to f=ua